sdk/src/main/java/io/radar/sdk/RadarActivityLifecycleCallbacks.kt

```diff
@@ -22,6 +26,11 @@ internal class RadarActivityLifecycleCallbacks : Application.ActivityLifecycleCa
         }
         count++
         foreground = count > 0
+
+        if (ContextCompat.checkSelfPermission(activity.applicationContext, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_DENIED &&
+            !ActivityCompat.shouldShowRequestPermissionRationale(activity, Manifest.permission.ACCESS_FINE_LOCATION)) {
```

Comment:
might've missed the nuance for this second condition, but is this preventing against the case where the app is first requesting permissions? maybe a comment above this condition would help clarify